src/main/java/net/openhft/chronicle/wire/JSONWire.java

```diff
@@ -373,6 +395,30 @@ public void copyOne(@NotNull WireOut wire, boolean inMap, boolean topLevel) thro
         throw new IORuntimeException("Unexpected chars '" + bytes.parse8bit(StopCharTesters.CONTROL_STOP) + "'");
     }
 
+    static boolean compareRest(@NotNull StreamingDataInput<?> in, @NotNull Bytes<?> s)
+            throws BufferUnderflowException, ClosedIllegalStateException {
+        if (s.length() > in.readRemaining())
+            return false;
+        long position = in.readPosition();
+        for (int i = 0; i < s.length(); i++) {
+            if (in.readUnsignedByte() != s.charAt(i)) {
+                in.readPosition(position);
+                return false;
+            }
+        }
+        int ch = in.peekUnsignedByte();
+        if (Character.isLetterOrDigit(ch)) {
+            in.readPosition(position);
+            return false;
+        }
+        while (ch > 0 && ch <= ' ') {
```

Comment:
Q: if this is going to return true by this point why does the iteration continue?

Reply:
ch = on line 416 takes the next character